cognite/client/_api/functions.py

```diff
@@ -964,9 +964,9 @@ def list(
             try:
                 IdentifierSequence.load(ids=function_id, external_ids=function_external_id).assert_singleton()
             except ValueError:
-                raise AssertionError(
+                raise ValueError(
                     "Both 'function_id' and 'function_external_id' were supplied, pass exactly one or neither."
-                )
+                ) from None
```

Comment:
Why from None here and not the Value Error above? Is it a syntactic sugar?

Reply:
Because the wording in one of the two different errors from`assert_singleton`:
```
1) ValueError: Exactly one of id or external id must be specified
2) ValueError: No identifiers specified
```
...mentions id + external id instead of what the user expects, function_id and function_external_id.

We need to refactor the Identifier stuff to more easily work with different identifiers, like user_identifier, name, key to name a few...